repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -465,7 +473,13 @@ public EntityMutationResponse mapAttributesAndClassifications(EntityMutationCont
                     reqContext.cache(updatedEntity);
 
                     if (DEFERRED_ACTION_ENABLED) {
-                        deleteDelegate.getHandler().createAndQueueTask(CLASSIFICATION_ONLY_PROPAGATION_DELETE, reqContext.getDeletedEdgesIds());
+                        List<AtlasEdge> classificationEdges = getAllClassificationEdges(vertex);
+                        for (AtlasEdge classificationEdge : classificationEdges) {
+                            String classificationVertexId = classificationEdge.getInVertex().getIdForDisplay();
+                            for (String deletedEdgeId: reqContext.getDeletedEdgesIds()) {
+                                deleteDelegate.getHandler().createAndQueueTask(CLASSIFICATION_ONLY_PROPAGATION_DELETE, deletedEdgeId, classificationVertexId);
+                            }
+                        }
```

Comment:
I see this logic is repeated multiple times, please check if it is possible to have separate method to reuse

Reply:
addressed